pkg/dockerregistry/server/util.go

```diff
@@ -84,4 +84,7 @@ func RememberLayersOfImageStream(ctx context.Context, cache cache.RepositoryDige
 	for dgst := range layers.Blobs {
 		_ = cache.AddDigest(digest.Digest(dgst), cacheName)
 	}
+	for dgst := range layers.Images {
+		_ = cache.AddDigest(digest.Digest(dgst), cacheName)
+	}
```

Comment:
dumb question but since the manifest is just another blob, why isn't it part of the layers.Blobs iteration already?

Reply:
Actually.... that's a good question.

Reply:
You're right, also, the API for ImageLayers is wrong.  Opening an origin PR to resolve the internal problem and fix the code issue, then I'll change the external API to clarify.

Once that's done we can then close this.

Reply:
Moved to https://github.com/openshift/origin/pull/20458.  That's the first fix.

Will change the public API in a subsequent PR.